src/FrameTen.js

```diff
@@ -0,0 +1,55 @@
+FrameTen = function() {
+  this.pinsRemaining = 10
+  this.rollOneScore = 0
+  this.rollTwoScore = 0
+  this.rollThreeScore = 0
+  this.rollOneDone = false
+  this.rollTwoDone = false
+  this.rollThreeDone = false
+};
+
+FrameTen.prototype.getRollOne = function(roll, player) {
+  if(roll === 10) {
+    this.rollOneScore = 10
+    player.newFrame()
+  }else if(roll !== 10) {
+    this.rollOneScore = this.getScore(roll)   
+  }
+  this.rollOneDone = true;
+};
+
+FrameTen.prototype.getRollTwo = function(roll, player){
+  if(roll === 10) {
+    this.rollTwoScore = 10
+    player.newFrame()
+  }else if(roll !== 10) {
+    this.rollTwoScore = this.getScore(roll)   
+  }; 
+  if(this.rollOneScore + this.rollTwoScore === 10){
+    player.newFrame;
+  }
+  this.rollTwoDone = true;
+};
+
```

Comment:
multiple conditionals - too many 'if's.

refactor by using a data structure to hold state - could be an object or hash.